x/staking/keeper/val_state_change_test.go

```diff
@@ -0,0 +1,25 @@
+package keeper_test
+
+import (
+	"github.com/line/lbm-sdk/simapp"
+	sdk "github.com/line/lbm-sdk/types"
+	"github.com/line/lbm-sdk/x/staking/teststaking"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestUnbondingToUnbondedPanic(t *testing.T) {
+	_, app, ctx := createTestInput()
+
+	//create a validator
+	addrDels := simapp.AddTestAddrs(app, ctx, 1, sdk.NewInt(0))
+	addrVals := simapp.ConvertAddrsToValAddrs(addrDels)
+
+	validator := teststaking.NewValidator(t, addrVals[0], PKs[0])
+	app.StakingKeeper.SetValidatorByConsAddr(ctx, validator)
+
+	// unbond the validator
+	require.Panics(t, func() {
+		app.StakingKeeper.UnbondingToUnbonded(ctx, validator)
+	})
```

Comment:
nit: it would be better to write test cases more clearly. Like, show all panic cases to try transit from (bonded/unbonded) to unbonded states.

Reply:
I changed test code to check transition on both side.
Please have a look again.